linker/shared/src/test/scala/org/scalajs/linker/MaxModuleSplittingTest.scala

```diff
@@ -66,4 +73,48 @@ class MaxModuleSplittingTest {
       assertEquals(expectedFiles, outputDirectory.fileNames().toSet)
     }
   }
+
+  @Test // #4542
+  def noMemExplostionOnImportChain(): AsyncResult = await {
+    /* Create a long import chain. Previous versions of the linker would attempt
+     * to construct the superset of import hops.
+     *
+     * In this test case, we construct 100 dynamics imports in a chain.
+     * Therefore, if the superset were calculated, it would occupy at least
+     * (assuming 1 byte per import hop) 2^100 Bytes = 2^50 PB.
```

Comment:
```suggestion
     * In this test case, we construct 100 dynamic imports in a chain.
     * Therefore, if the power set were calculated, it would occupy at least
     * (assuming 1 byte per import hop) 2^100 Bytes = 2^50 PB.
```